src/org/ods/orchestration/usecase/JiraUseCase.groovy

```diff
@@ -464,4 +474,131 @@ class JiraUseCase {
+        def jqlQuery = [
+            fields: fields,
+            jql: jql,
+            expand: []
+        ]
+
+        return jira.getIssuesForJQLQuery(jqlQuery) ?: []
+    }
+
+    List createSecurityVulnerabilityIssues(List aquaCriticalVulnerabilityRepos) {
+        def securityVulnerabilityIssueKeys = [];
+        try {
+            for (def repo : aquaCriticalVulnerabilityRepos) {
+                def jiraComponentId = getJiraComponentId(repo)
+                for (def vulnerability : repo.data.openshift.aquaCriticalVulnerability) {
+                    def vulerabilityMap = vulnerability as Map
+                    def issueKey = createOrUpdateSecurityVulnerabilityIssue(
+                        vulerabilityMap.name,
+                        jiraComponentId,
+                        buildSecurityVulnerabilityIssueDescription(
+                            vulerabilityMap,
+                            repo.data.openshift.gitUrl,
+                            repo.data.openshift.gitBranch,
+                            repo.data.openshift.repoName,
+                            repo.data.openshift.nexusReportLink))
+                    securityVulnerabilityIssueKeys.add(issueKey)
+                }
+            }
+        } catch (JiraNotPresentException e) {
+            logger.warn(e.getMessage())
+            return []
+        }
+        return securityVulnerabilityIssueKeys
+    }
+
+    String createOrUpdateSecurityVulnerabilityIssue(String vulnerabilityName, String jiraComponentId,
+                                                    String description) {
+        if (!jira) {
+            throw new JiraNotPresentException("JiraUseCase not present, cannot create security vulnerability issue.")
+        }
+
+        def issueSummary =  SECURITY_VULNERABILITY_ISSUE_SUMMARY.replace(VULNERABILITY_NAME_PLACEHOLDER,
+            vulnerabilityName)
+
+        def fixVersion = null
+        if (project.isVersioningEnabled) {
+            fixVersion = project.getVersionName()
+        }
+        def fullJiraComponentName = JIRA_COMPONENT_TECHNOLOGY_PREFIX + jiraComponentId
+
+        List securityVulnerabilityIssues = loadJiraSecurityVulnerabilityIssues(issueSummary,
+            fixVersion, fullJiraComponentName, project.jiraProjectKey)
+        if (securityVulnerabilityIssues?.size() >= 1) { // Transition the issue to "TO DO" state
+            transitionIssueToToDo(securityVulnerabilityIssues.get(0).id)
+            return (securityVulnerabilityIssues.get(0) as Map)?.key
+        } else { // Create the issue
+            return (createIssueTypeSecurityVulnerability(fixVersion: fixVersion, component: fullJiraComponentName,
+                priority: SECURITY_VULNERABILITY_ISSUE_PRIORITY, projectKey: project.jiraProjectKey,
+                summary: issueSummary, description: description)
+                as Map)?.key
+        }
+    }
+
+    String buildSecurityVulnerabilityIssueDescription(Map vulnerability, String gitUrl, String gitBranch,
```

Comment:
Add to thiss method the @NonCPS annotation and check the rest (to add it if is suitable).